### PR TITLE
feat(metrics): add the engine type to the metrics sent

### PR DESCRIPTION
**Pull request by emjin**
This adds the engine that the user requests to the metrics sent. Otherwise, for CI scans this information is not present anywhere.

This also moves information that can be sent earlier to be sent earlier so that even in case of failure we have it.

Test plan: see added tests

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by emjin** [comment]
Do I need to update the docs?

**Comment by ievans** [comment]
@emjin concurrent update to PRIVACY.md / docs as well?

File: cli/tests/e2e/snapshots/test_metrics/test_metrics_payload/metrics-payload.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "environment": {
     "ci": null,
     "configNamesHash": "d3e03c9938537c0a0668cec7204f37f1e9da87d03e18e8cc444dd7454b36c4d5",
+    "engineType": "OSS",
```
Comment (brendongo) [inline cli/tests/e2e/snapshots/test_metrics/test_metrics_payload/metrics-payload.json]: why is this engineType not engineRequested?

Comment (brandonspark) [reply cli/tests/e2e/snapshots/test_metrics/test_metrics_payload/metrics-payload.json]: i have the same question. this is weird, wonder why it's there

Comment (emjin) [reply cli/tests/e2e/snapshots/test_metrics/test_metrics_payload/metrics-payload.json]: Oh my bad that file shouldn't be there

File: PRIVACY.md [hunk 2]
```diff
@@ -181,6 +182,7 @@ r2c will:
...
 | Value       |                                         |                                                                        |                                                                                            |                                                                                                                                                                                       |                |
+|             | Engine requested                        | The engine requested by the user                                       | Understand what engines people are using and debug problems                                | `Interfile`                                                                                                                                                                           | str            |
```
Comment (brandonspark) [inline PRIVACY.md]: i'm not in love with this description (on the far right column), i'd prefer some mention of "debug engine-specific problems", perhaps

File: PRIVACY.md [hunk 3]
```diff
@@ -236,6 +238,12 @@ The Semgrep team uses this to answer the following questions:
   We use this to evaluate experimental features
   and understand their production-readiness.
 
+> Engine requested (OSS, Pro, Interfile)
```
Comment (brandonspark) [inline PRIVACY.md:241]: same sort of deal here. i think i just regard "what engines people are using" as rather casual language

**Review by brandonspark — APPROVED** [review]
Let's Go To Mars!

**Review by brendongo — APPROVED** [review]

**Pull request merged**